File: src/2025-03-04-12-20-21--deepseek-vr-0304/javascript/exercises/practice/beer-song/beer-song.js
//
// This is only a SKELETON file for the 'Beer Song' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const recite = (initialBottlesCount, takeDownCount) => {
  const verses = [];
  let current = initialBottlesCount;
  for (let i = 0; i < takeDownCount; i++) {
    if (i > 0) {
      verses.push('');
    }
    const [line1, line2] = generateVerse(current);
    verses.push(line1, line2);
    current = current === 0 ? 99 : current - 1;
  }
  return verses;
};

function generateVerse(current) {
  let line1, line2;
  if (current === 0) {
    line1 = 'No more bottles of beer on the wall, no more bottles of beer.';
    line2 = 'Go to the store and buy some more, 99 bottles of beer on the wall.';
  } else {
    const bottles = current === 1 ? '1 bottle' : `${current} bottles`;
    line1 = `${bottles} of beer on the wall, ${bottles} of beer.`;
    const next = current - 1;
    if (next === 0) {
      line2 = 'Take it down and pass it around, no more bottles of beer on the wall.';
    } else {
      const nextBottles = next === 1 ? '1 bottle' : `${next === 0 ? 'no more' : next} bottles`;
      const action = current === 1 ? 'it' : 'one';
      line2 = `Take ${action} down and pass it around, ${nextBottles} of beer on the wall.`;
    }
  }
  return [line1, line2];
}

File: src/2025-03-04-12-20-21--deepseek-vr-0304/python/exercises/practice/beer-song/beer_song.py
def recite(start, take=1):
    numbers = []
    current = start
    for _ in range(take):
        numbers.append(current)
        next_num = current - 1
        if next_num < 0:
            next_num = 99
        current = next_num

    verses = []
    for i, n in enumerate(numbers):
        if n == 0:
            line1 = "No more bottles of beer on the wall, no more bottles of beer."
            line2 = "Go to the store and buy some more, 99 bottles of beer on the wall."
        elif n == 1:
            line1 = "1 bottle of beer on the wall, 1 bottle of beer."
            line2 = "Take it down and pass it around, no more bottles of beer on the wall."
        else:
            line1 = f"{n} bottles of beer on the wall, {n} bottles of beer."
            next_bottle = n - 1
            if next_bottle == 1:
                line2 = "Take one down and pass it around, 1 bottle of beer on the wall."
            else:
                line2 = f"Take one down and pass it around, {next_bottle} bottles of beer on the wall."

        verses.append(line1)
        verses.append(line2)
        if i != len(numbers) - 1:
            verses.append("")

    return verses
